### PR TITLE
Surreal work

### PR DESCRIPTION
**Pull request by sctfn**
So, I made a mistake when proving a fundamental theorem of surreals a few years ago, and the consequences finally caught up with me. I fixed the theorem (noeta) and proved some consequences

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -522922,6 +522938,70 @@ Real and complex numbers (cont.)
...
+
+  ${
+    $d F x $.  $d A x $.  $d B x $.
+    $( The image of a function under a singleton.  (Contributed by Scott
+       Fenton, 8-Aug-2024.) $)
+    fvimasn $p |- ( ( F Fn A /\ B e. A ) -> ( F " { B } ) = { ( F ` B ) } ) $=
+      ( vx wfn wcel wa csn cima cv wbr cab wceq imasng adantl velsn eqcom bitri
+      cfv fnbrfvb syl5rbb abbi1dv eqtrd ) CAEZBAFZGZCBHIZBDJZCKZDLZBCSZHZUEUGUJ
+      MUDDBACNOUFUIDULUHULFZUKUHMZUFUIUMUHUKMUNDUKPUHUKQRABUHCTUAUBUC $.
+  $}
```
Comment (tirix) [inline set.mm:522957]: It appears this is the same as [`~fnsnfv`](https://us.metamath.org/mpeuni/fnsnfv.html), just commuted.

File: set.mm [hunk 2]
```diff
@@ -522922,6 +522938,70 @@ Real and complex numbers (cont.)
...
+    ( cres cdm cin dmres inss1 eqsstri ) ABCDBADZEBABFBIGH $.
+
+
+  $( The union of two ordinals is an ordinal.  (Contributed by Scott Fenton,
+     9-Aug-2024.) $)
+  onun2 $p |- ( ( A e. On /\ B e. On ) -> ( A u. B ) e. On ) $=
+    ( con0 wcel wa wss wceq ssequn1 wi eleq1a adantl syl5bi ssequn2 adantr word
+    cun wo eloni ordtri2or2 syl2an mpjaod ) ACDZBCDZEZABFZABPZCDZBAFZUEUFBGZUDU
+    GABHUCUIUGIUBBCUFJKLUHUFAGZUDUGBAMUBUJUGIUCACUFJNLUBAOBOUEUHQUCARBRABSTUA
+    $.
```
Comment (tirix) [inline set.mm:523004]: Similarly, [`~onun2i`](https://us.metamath.org/mpeuni/onun2i.html) can be proved using this one.

File: set.mm [hunk 3]
```diff
@@ -529082,6 +530393,111 @@ property of surreals and will be used (via surreal cuts) to prove many
...
+  c0s $a class 0s $.
+
+  $( Declare the class syntax for surreal one. $)
+  c1s $a class 1s $.
+
+  $( Define surreal zero.  This is the simplest cut of surreal number sets.
+     (Contributed by Scott Fenton, 7-Aug-2024.) $)
+  df-0s $a |- 0s = ( (/) |s (/) ) $.
+
+  $( Define surreal one.  This is the simplest number greater than zero.
```
Comment (benjub) [inline set.mm:530415]:
```suggestion
  $( Define surreal one.  This is the simplest surreal number greater than surreal zero.
```

File: set.mm [hunk 4]
```diff
@@ -529120,21 +530536,21 @@ property of surreals and will be used (via surreal cuts) to prove many
...
     $( Define the left options of a surreal.  This is the set of surreals that
-       are "closest" on the left to the given surreal.  (Contributed by Scott
-       Fenton, 17-Dec-2021.) $)
+       are simpler and less than the given surreal.  (Contributed by Scott
+       Fenton, 6-Aug-2024.) $)
     df-left $a |- _L = ( x e. No |-> { y e. ( _Old ` ( bday ` x ) ) |
-     A. z e. No ( ( y <s z /\ z <s x ) -> ( bday ` y ) e. ( bday ` z ) ) } ) $.
+     y <s x } ) $.
 
     $( Define the left options of a surreal.  This is the set of surreals that
```
Comment (benjub) [inline set.mm:530549]:
```suggestion
    $( Define the right options of a surreal.  This is the set of surreals that
```

File: set.mm [hunk 5]
```diff
@@ -529175,16 +530591,375 @@ property of surreals and will be used (via surreal cuts) to prove many
...
+     Fenton, 8-Aug-2024.) $)
+  new0 $p |- ( _N ` (/) ) = { 0s } $=
+    ( c0 cnew cfv cmade cold cdif c0s csn con0 wcel wceq 0elon ax-mp made0 old0
+    newval difeq12i dif0 3eqtri ) ABCZADCZAECZFZGHZAFUDAIJTUCKLAPMUAUDUBANOQUDR
+    S $.
+
+  ${
+    $d A x l r $.  $d B x l r $.
+    $( If ordinal ` A ` is less than or equal to ordinal ` B ` , then their
+       made sets are subsets.  (Contributed by Scott Fenton, 7-Aug-2024.) $)
```
Comment (benjub) [inline set.mm:530818]:
```suggestion
    $( If ordinal ` A ` is less than or equal to ordinal ` B ` , then the made set
    of ` A ` is included in the made set of ` B ` .
    made sets are subsets.  (Contributed by Scott Fenton, 7-Aug-2024.) $)
```

File: set.mm [hunk 6]
```diff
@@ -529175,16 +530591,375 @@ property of surreals and will be used (via surreal cuts) to prove many
...
+      syld ) AFGZBFGZABHZUAZCIZDIZUBUCVGVHUDUEEIZJKZDLAMZNZOZPZCVMPZEQRZVJDLBMZ
+      NZOZPZCVSPZEQRZALTZBLTZVFVOWASZEQUMVPWBHVFWEEQVFVIQGZKZVOVNCVSPZWAWGVMVSH
+      ZVOWHSWGVLVRWGVKVQVFVKVQHZWFVEVCWJVDABLUFUGUHUIUJZVNCVMVSUKULWGVNVTCVSWGW
+      IVNVTSWKVJDVMVSUKULUNVBUOVOWAEQUPUQVCVDWCVPJVEEACDURUSVDVCWDWBJVEEBCDURUT
+      VA $.
+  $}
+
+  ${
+    $d A x b $.
+    $( The older than set is a subset of the made set.  (Contributed by Scott
```
Comment (benjub) [inline set.mm:530833]:
```suggestion
    $( The older-than set is a subset of the made set.  (Contributed by Scott
```

File: set.mm [hunk 7]
```diff
@@ -528259,134 +529263,326 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
...
+      wceq simpl2 simpl3 simpr noinfbnd1 eqbrtrd con0 noetainflem1 adantr simp2
+      wi sselda noinfno nodmon syl sltres mpd ) EMNZFOUAZFMNZUBZJFNZUCZIGUDZPZJ
+      VIPZQRZIJQRZVHVJGVKQVHVDVEVJGUFVCVDVEVGUGZVCVDVEVGUHZABCDEFGHIKLUESVHVDVE
+      VGGVKQRVNVOVFVGUIABCDFGJHMKUJTUKVHIONZJONVIULNZVLVMUPVFVPVGABCDEFGHIKLUMU
+      NVFFOJVCVDVEUOUQVHGONZVQVHVDVEVRVNVOABCDFGHMKURSGUSUTIJVIVATVB $.
+
+    $d A a $.  $d a b $.  $d a g $.  $d a p $.  $d a q $.  $d a x $.  $d B a $.
+    $d B b $.  $d b g $.  $d b v $.  $d b x $.  $d b y $.  $d p q $.  $d T b $.
+    $d T g $.  $d T q $.  $d W p $.  $d W q $.
+    $( Lemma for ~ noeta .  If ` A ` preceeds ` B ` , then ` W ` is greater
```
Comment (benjub) [inline set.mm:529371]:
```suggestion
    $( Lemma for ~ noeta .  If ` A ` precedes ` B ` , then ` W ` is greater
```

**Comment by tirix** [comment]
Nice! So your original version of `noeta` was actually not wrong, but too weak, right?
This translated also in weaker statements for `etasslt` and `scutbdaybnd`, which I assume were too weak to prove `madebdayim` and `oldbdayim`.

File: set.mm [hunk 8]
```diff
@@ -528259,134 +529263,326 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
...
        bound on the birthday of that surreal.  Alling's axiom FE. (Contributed
-       by Scott Fenton, 7-Dec-2021.) $)
-    noeta $p |- ( ( ( A C_ No /\ A e. V ) /\ ( B C_ No /\ B e. W ) /\
-            A. x e. A A. y e. B x <s y ) ->
+       by Scott Fenton, 9-Aug-2024.) $)
+    noeta $p |- ( ( ( ( A C_ No /\ A e. V ) /\ ( B C_ No /\ B e. W ) /\
+            A. x e. A A. y e. B x <s y ) /\
+            ( O e. On /\ ( bday " ( A u. B ) ) C_ O ) ) ->
           E. z e. No ( A. x e. A x <s z /\ A. y e. B z <s y /\
-                       ( bday ` z ) C_ suc U. ( bday " ( A u. B ) ) ) ) $=
```
Comment (benjub) [inline set.mm:528376]: Since you added an antecedent in the formalization, you should also add that condition in the comment; something like
> ... and the birthdays of members of A and B are bounded, there is a surreal lying strictly between the two sets, whose birthday is bounded by the successor of the union of the set of birthdays of the surreals in these sets.

I'm not familiar enough with surreals, so you'll find the correct formulation.

**Comment by sctfn** [comment]
Precisely! Also, as I reread the references in the literature, this is
closer to the cited version of axiom FE in Alling

File: set.mm (hunk 1 above)
Comment (sctfn) [reply set.mm]: Ok, theorem is removed

File: set.mm (hunk 2 above)
Comment (sctfn) [reply set.mm]: Done

File: set.mm [hunk 9]
```diff
@@ -522922,6 +522938,70 @@ Real and complex numbers (cont.)
...
+     by Scott Fenton, 7-Aug-2024.) $)
+  rexbi $p |- ( A. x e. A ( ph <-> ps ) ->
+  ( E. x e. A ph <-> E. x e. A ps ) ) $=
+    ( wb wral cv wcel wa wex wrex wal wi df-ral pm5.32 biimpi alimi exbi df-rex
+    sylbi syl 3bitr4g ) ABEZCDFZCGDHZAIZCJZUEBIZCJZACDKBCDKUDUFUHEZCLZUGUIEUDUE
+    UCMZCLUKUCCDNULUJCULUJUEABOPQTUFUHCRUAACDSBCDSUB $.
+
+  ${
+    $d F x $.  $d A x $.  $d B x $.
+    $( The image of a function under a singleton.  (Contributed by Scott
```
[the remark this answers is not in the document]
Comment (sctfn) [reply set.mm]: Both done

File: set.mm [hunk 10]
```diff
@@ -522922,6 +522938,70 @@ Real and complex numbers (cont.)
     UAEZVKBTQEZEZVLBTNDZEZEZVNBCFVPVMEZVQVNEVOQEVRNPGVOQTHIVPVMBHIIVMUAEZVNVLEU
     MVSUNQRTHIVMUABHIIUGVLVKEUOUARBHIIBRKIURRAHIIARKII $.
 
+  $( Distribute restricted quantification over a biconditional.  (Contributed
+     by Scott Fenton, 7-Aug-2024.) $)
+  rexbi $p |- ( A. x e. A ( ph <-> ps ) ->
+  ( E. x e. A ph <-> E. x e. A ps ) ) $=
+    ( wb wral cv wcel wa wex wrex wal wi df-ral pm5.32 biimpi alimi exbi df-rex
+    sylbi syl 3bitr4g ) ABEZCDFZCGDHZAIZCJZUEBIZCJZACDKBCDKUDUFUHEZCLZUGUIEUDUE
+    UCMZCLUKUCCDNULUJCULUJUEABOPQTUFUHCRUAACDSBCDSUB $.
```
[the remark this answers is not in the document]
Comment (sctfn) [reply set.mm]: Done and basic shortening is done too

File: set.mm [hunk 11]
```diff
@@ -528671,6 +529873,25 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
         |^| ( bday " { y e. No | ( a <<s { y } /\ { y } <<s b ) } ) ) ) $.
   $}
 
+  ${
+    $d A x y z $.  $d B x y z $.
+    $( A version of ~ noeta with less hypotheses but a weaker upper bound
```
[the remark this answers is not in the document]
Comment (sctfn) [reply set.mm:529878]: Fixed

File: set.mm [hunk 12]
```diff
@@ -522922,6 +522938,70 @@ Real and complex numbers (cont.)
...
+    snres0 $p |- ( ( { <. A , B >. } |` C ) = (/) <-> -. A e. C ) $=
+      ( cop csn cres c0 wceq cdm cin wcel wn wrel wb relres reldm0 ax-mp dmsnop
+      dmres ineq2i eqtri eqeq1i disjsn 3bitri ) ABEFZCGZHIZUGJZHIZCAFZKZHIACLMU
+      GNUHUJOUFCPUGQRUIULHUICUFJZKULUFCTUMUKCABDSUAUBUCCAUDUE $.
+  $}
+
+  $( Subset relationship for the domain of a restriction.  (Contributed by
+     Scott Fenton, 9-Aug-2024.) $)
+  resdmss $p |- dom ( A |` B ) C_ B $=
+    ( cres cdm cin dmres inss1 eqsstri ) ABCDBADZEBABFBIGH $.
```
[the remark this answers is not in the document]
Comment (sctfn) [reply set.mm]: Done

File: set.mm [hunk 13]
```diff
@@ -528951,41 +530213,90 @@ property of surreals and will be used (via surreal cuts) to prove many
...
+      OWPWKXPWTWNQIZTWSWNPKZXEFWNOZEBOZLWOXPWTXQVTWTWAXOXJSWEUQZURXPWSXRXTVTWSW
+      AXOXISXNXRXMWLWEPUSUTZXPWGXTWGWJWKXNXMVAXSWFEBXEWFFWEAVBZWHWEWDNVCVEVFVGR
+      EFBWNVHVIXPXQXCTXRXBXFEWNOZLWPXPXCXQVTXCWAXOXLSYAVJXPXRXBYDYBVTXBWAXOXKSX
+      PWJYDWGWJWKXNXMVKXFWJEWEYCEAVOXEWIFCWDWEWHNVPVLVEVGREFWNCVHVIWGWJWKXNXMVM
+      RVNVQVRVS $.
+  $}
+
+  ${
+    $d A x y z $.  $d B x y z $.
+    $( A version of ~ etasslt with less hypotheses but a looser upper bound.
```
[the remark this answers is not in the document]
Comment (sctfn) [reply set.mm:530238]: This is more terminology around ordinals in this case. I accepted your suggestion, though

File: set.mm (hunk 3 above)
Comment (sctfn) [reply set.mm:530415]: Done

File: set.mm (hunk 4 above)
Comment (sctfn) [reply set.mm]: Yeah, done 😁

File: set.mm (hunk 5 above)
Comment (sctfn) [reply set.mm]: Fixed

File: set.mm (hunk 6 above)
Comment (sctfn) [reply set.mm:530833]: Done

File: set.mm [hunk 14]
```diff
@@ -29631,7 +29639,7 @@ generally appear in a single form (either definitional, but more often
     $( Inference adding restricted existential quantifier to both sides of an
        equivalence.  (Contributed by NM, 26-Oct-1999.) $)
     rexbiia $p |- ( E. x e. A ph <-> E. x e. A ps ) $=
-      ( cv wcel pm5.32i rexbii2 ) ABCDDCFDGABEHI $.
+      ( wb wrex rexbi mprg ) ABFACDGBCDGFCDABCDHEI $.
```
Comment (benjub) [inline set.mm:29642]: Looks like a lengthening.

Comment (sctfn) [reply set.mm]: You're right. Restored old proof

**Comment by david-a-wheeler** [comment]
> So, I made a mistake when proving a fundamental theorem of surreals a few years ago, and the consequences finally caught up with me. I fixed the theorem (noeta) and proved some consequences.

I hope you continue to make such awesome "mistakes" :-).

File: set.mm [hunk 15]
```diff
@@ -527453,6 +527584,45 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
       JYRUXKUWEUYMUXKUWBUUPUWCUUSUWDUVTCAWRUVTCBWRYJYKUJYSUUAUUBUUCUUDVJUURUVAU
       VBUUTSZUUTUVAUVBSZUURUUHVUDUWLBCUUEUHVUDUVBUUTUVASVUEUVAUVBUUTYTUVBUUTUVA
       YTRWDUUF $.
+
+    $( Given ` A ` less than ` B ` , equal to ` B ` up to ` X ` , and undefined
+       at ` X ` , then ` A ( X ) = 1o ` .  (Contributed by Scott Fenton,
+       9-Aug-2024.) $)
+    nogt01o $p |- ( ( ( A e. No /\ B e. No /\ X e. On ) /\
+    ( ( A |` X ) = ( B |` X ) /\ A <s B ) /\ ( B ` X ) = (/) ) ->
+    ( A ` X ) = 1o ) $=
```
Comment (icecream17) [inline set.mm:527593]: ` B ` is undefined at ` X ` rather than ` A `, so maybe reword (it also matches the label)...
```suggestion
    $( Given ` B ` greater than ` A ` , equal to ` A ` up to ` X ` , and
       undefined at ` X ` , then ` A ( X ) = 1o ` .  (Contributed by Scott
       Fenton, 9-Aug-2024.) $)
    nogt01o $p |- ( ( ( A e. No /\ B e. No /\ X e. On ) /\
    ( ( A |` X ) = ( B |` X ) /\ A <s B ) /\ ( B ` X ) = (/) ) ->
    ( A ` X ) = 1o ) $=
```

**Review by icecream17 — APPROVED** [review]

**Review by tirix — APPROVED** [review]

**Comment by sctfn** [comment]
Whoops. Got yours, missed @icecream17. Your comment is fixed in the current PR. I’ll get his in the next one. On Aug 12, 2024, at 6:19 AM, Benoit ***@***.***> wrote:﻿
@benjub commented on this pull request.



In set.mm:
>      $( The full-eta axiom for the surreal numbers.  This is the single most
        important property of the surreals.  It says that, given two sets of
        surreals such that one comes completely before the other, there is a
        surreal lying strictly between the two.  Furthermore, there is an upper
        bound on the birthday of that surreal.  Alling's axiom FE. (Contributed
-       by Scott Fenton, 7-Dec-2021.) $)
-    noeta $p |- ( ( ( A C_ No /\ A e. V ) /\ ( B C_ No /\ B e. W ) /\
-            A. x e. A A. y e. B x <s y ) ->
+       by Scott Fenton, 9-Aug-2024.) $)
+    noeta $p |- ( ( ( ( A C_ No /\ A e. V ) /\ ( B C_ No /\ B e. W ) /\
+            A. x e. A A. y e. B x <s y ) /\
+            ( O e. On /\ ( bday " ( A u. B ) ) C_ O ) ) ->
           E. z e. No ( A. x e. A x <s z /\ A. y e. B z <s y /\
-                       ( bday ` z ) C_ suc U. ( bday " ( A u. B ) ) ) ) $=

@sctfn did you miss this one (and also @icecream17's comment below) ?
